src/defaultapps/terminal/terminal.js

```diff
@@ -1,100 +1,40 @@
-    }
-    process(cont)
-    {
-       let string=cont.value;
-       //let opt=commandLineProcessor(string);
-       let lex=string.split(' ');
-       var holder=this.state.lineHolder;
-       if(lex[0]==="ls")
-       {
-           
-           var content=this.state.directory.folderContents;
-           for(let dir in content)
-           {
-               holder.push(
-                <div key={this.state.id+this.lines} className="line">
-                <div className="line">{content[dir].folderName}/</div>
-                </div>
-               );
-           }
-           this.setState({lineHolder:holder});
-       }
-       else if(lex[0]==="cd")
-       {
-           if(lex[1]==="..")
-           {
-                //TO be done
-           }
-           else
-           {
-            let content=this.state.directory.folderContents;
-            this.setState({directory:content[lex[1]]});
-           }
-       }
-       else
-       {
-           holder.push(
-            <div key={this.state.id+this.lines} className="line">
-            <div className="line" style={{color:'red'}}>{lex.join(' ')} is not a valid command</div>
-            </div>
-           )
-       }
-    }
-    enterHouseKeeping=(e)=>{
-        if(e.key === "Enter")
-        {
-            e.preventDefault();
-            var prevLine=document.getElementById(this.state.id+'line'+(this.lines-1));
-            this.process(prevLine);
-            this.newLine();
-        }
-    }
-    newLine()
-    {
-        console.log('yolo');
-        let holder=this.state.lineHolder;
-        console.log(this.props.directory.folderName)
-        holder.push(
-            <div key={this.state.id+this.lines} className="line">
-                <div className="directory">{this.state.directory.folderName}$</div>
-                <input id={this.state.id+'line'+this.lines} className="type" onKeyPress={this.enterHouseKeeping} autofocus="true"></input>
-            </div>
-        );
-        this.setState({lineHolder:holder},()=>{
-        if(this.lines!==0)
-        {
-            var prevLine=document.getElementById(this.state.id+'line'+(this.lines-1));
-            prevLine.disabled=true;
-            var nextLine=document.getElementById(this.state.id+'line'+(this.lines));
-            nextLine.focus();
+    
+    handleKeyPress = (e) => {
+        if(e.key !== 'Enter'){
+            if(e.key === 'Backspace'){ 
+                this.state.cmd = this.state.cmd.slice(0,this.state.cmd.length -1)
+            }else{
+             this.state.cmd = this.state.cmd.concat(e.key)
+            }
+        }else{
+            if(this.state.cmd.length===0){
+                //new line will be implemented
+            }else
+            console.log(this.state.cmd)
         }
-        this.lines++;
-        });
     }
     render()
-    {
+    {       
         return(
             <div id="terminal">
-              {this.state.lineHolder.map(line=>line)}  
+                <div id="bash">
```

Comment:
hmm id's like terminal and bash should not be used. When opening multiple terminal windows the css probably gets messed up. More robust thing would be bash+{rand_id}. Probably look into this.

Reply:
I just checked it, there is no problem with the CSS. 
I could however change `id="bash"` to `id = "terminal-input"`

Reply:
Just to be sure :) cause css doesnt like 2 div's with same id's